packages/local-environment/src/domain.ts

```diff
@@ -41,27 +49,33 @@ export class NomadDomain {
       watcher: signer,
       recoveryManager: signer,
     });
+
+  }
+
+  static newHardhatNetwork(name: string, domain: number, creationParams?: DomainOptions): NomadDomain {
+    const network = new HardhatNetwork(name, domain, creationParams?.forkurl, creationParams?.weth, creationParams?.docker);
+    return new NomadDomain(network, creationParams?.nomadEnv);
   }
 
-  addNomadEnv(e: NomadEnv) {
+  addNomadEnv(e: NomadEnv): void {
     this.nomadEnv = e;
   }
 
   get name(): string {
     return this.network.name;
   }
 
-  ensureAgents(metricsPort = 9090) {
-    if (!this.agents) this.agents = new Agents(this, metricsPort, this.nomadEnv);
+  ensureAgents(metricsPort = 9090, docker?: Dockerode): void {
+    if (!this.agents) this.agents = new Agents(this, metricsPort, docker, this.nomadEnv);
   }
 
-  connectNetwork(d: NomadDomain) {
+  connectNetwork(d: NomadDomain): void {
     if (!this.connections().includes(d.network.name))
       this.connectedNetworks.push(d);
   }
```

Comment:
Maybe add a parameter `connectInReverse`:
```
  connectNetwork(d: NomadDomain, connectInReturn=true): void {
    if (!this.connections().includes(d.network.name)) {
      this.connectedNetworks.push(d);
      if (connectInReturn) {
         d.connectInReturn(this, false)
      }
    }
  }
```
Question to think: how will it help us?

Reply:
Reduces the redundancy in code that forces us to do connectNetwork twice for each domain. Implementing.